File: packages/js/core/src/manifest/formats/web3api.meta/migrators/0.0.1-prealpha.1_to_0.0.1-prealpha.3.ts
/* eslint-disable @typescript-eslint/naming-convention */

import { MetaManifest as OldManifest } from "../0.0.1-prealpha.1";
import { MetaManifest as NewManifest } from "../0.0.1-prealpha.3";

export function migrate(old: OldManifest): NewManifest {
  const manifest: Record<string, unknown> = {
    ...old,
  };

  delete manifest.name;

  return {
    ...manifest,
    __type: "MetaManifest",
    format: "0.0.1-prealpha.3",
    displayName: old.name,
  };
}

File: packages/py/manifests/polywrap/src/formats/polywrap.meta/migrators/_0_0_1_prealpha_1_to__0_0_1_prealpha_3.py
from ... import import_manifest
from typing import TypeAlias
import os

local_path: os.PathLike = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))

__0_0_1_prealpha_1 = import_manifest('0.0.1-prealpha.1', local_path)
OldManifest: TypeAlias = __0_0_1_prealpha_1.MetaManifest
__0_0_1_prealpha_3 = import_manifest('0.0.1-prealpha.3', local_path)
NewManifest: TypeAlias = __0_0_1_prealpha_3.MetaManifest

def migrate(old: OldManifest) -> NewManifest:
    new: NewManifest = old
    new.__type = "MetaManifest"
    new.format = "0.0.1-prealpha.3"
    return new